tests/test_mcp_client.py

```
@@ -0,0 +1,249 @@
+                "Authorization": "Bearer secret",
+                "Mcp-Session-Id": "session-secret",
+                "X-Trace-Id": "trace-123",
+            }
+        )
+
+        self.assertEqual(sanitized["Authorization"], "[redacted]")
+        self.assertEqual(sanitized["Mcp-Session-Id"], "[redacted]")
+        self.assertEqual(sanitized["X-Trace-Id"], "trace-123")
+
+    def test_validate_mcp_http_url_blocks_localhost(self) -> None:
+        with self.assertRaises(McpClientError):
```

Ruff (0.15.20)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)